balancer-js/src/modules/sor/pool-data/onChainData.ts

```diff
@@ -197,7 +209,21 @@ export async function getOnChainBalances(
         }
       });
 
-      subgraphPools[index].totalShares = formatFixed(totalSupply, 18);
+      // Pools with pre minted BPT
+      if (
+        subgraphPools[index].poolType.includes('Linear') ||
+        subgraphPools[index].poolType === 'StablePhantom'
```

Comment:
Note: There is a new StablePhantom type coming out soon which this will also need to support. I'm not sure what the type is yet, @markusbkoch is working on it.

Reply:
Waiting for to see what the final contract name will be to keep it the same for consistency